optuna/study/_optimize.py

```diff
@@ -213,6 +213,9 @@ def _run_trial(
         # TODO(mamu): Handle multi-objective cases.
         state = TrialState.PRUNED
         func_err = e
+    except KeyboardInterrupt as e:
+        state = TrialState.FAIL
+        func_err = e
```

Comment:
I'm not sure that we should do to that but it might be better to set the result of `sys.exc_info()` to `func_err_fail exc_info` here as the following `Exception` handling branch does.
What do you think of that? > @CorentinNeovision @himkt

Reply:
Thank you for the catch. It makes sense. `except` can takes multiple exceptions so we can update here from

```python
    except KeyboardInterrupt as e:
        state = TrialState.FAIL
        func_err = e
    except Exception as e:
        state = TrialState.FAIL
        func_err = e
        func_err_fail_exc_info = sys.exc_info()
```

to

```python
    except (Exception, KeyboardInterrupt) as e:
        state = TrialState.FAIL
        func_err = e
        func_err_fail_exc_info = sys.exc_info()
```

:link: https://docs.python.org/3/tutorial/errors.html#handling-exceptions